webapp/packages/core-blocks/src/FormControls/Combobox.tsx

```diff
@@ -87,245 +83,152 @@ export const Combobox: ComboboxType = observer(function Combobox({
   iconSelector,
   titleSelector,
   isDisabled,
-  onChange = () => {},
   onSelect,
-  onSwitch,
   ...rest
 }: ControlledProps<any, any> | ObjectProps<any, any, any>) {
   const layoutProps = getLayoutProps(rest);
+  const inputId = useId();
   rest = filterLayoutFakeProps(rest);
   const translate = useTranslate();
   const context = useContext(FormContext);
-  const menuRef = useRef<HTMLDivElement>(null);
-  const [inputRef, setInputRef] = useState<HTMLInputElement | null>(null);
-  const styles = useS(comboboxStyles);
-
-  const menu = useMenuState({
-    placement: 'bottom-end',
-    currentId: null,
-    gutter: 4,
-    unstable_fixed: true,
-  });
-
-  const [searchValue, setSearchValue] = useState<string | null>(null);
-
-  const filteredItems = getComputed(() => {
-    const result = items.filter(item => !searchValue || valueSelector(item).toUpperCase().includes(searchValue.toUpperCase()));
-
-    if (isDisabled) {
-      return result.sort((a, b) => Number(isDisabled(a)) - Number(isDisabled(b)));
-    }
-
-    return result;
-  });
-
-  let value: string | number | readonly string[] | undefined = controlledValue ?? defaultValue ?? undefined;
 
+  let selectedKey: string | string[] = controlledValue ?? defaultValue ?? undefined;
   if (state && name !== undefined && name in state) {
-    value = state[name];
-  }
-
-  const selectedItem = items.find((item, index) => keySelector(item, index) === value);
-
-  let inputValue = (selectedItem ? valueSelector(selectedItem) : searchValue) ?? '';
-
-  if (searchValue !== null && selectedItem && valueSelector(selectedItem) !== searchValue) {
-    inputValue = searchValue;
+    selectedKey = state[name];
   }
 
-  const hideMenu = items.length === 1 && (!!selectedItem || isDisabled?.(items[0]) === true);
-
-  function handleClick() {
-    if (menu.visible) {
-      menu.hide();
-    } else {
-      menu.show();
-    }
-  }
-
-  const handleChange = useCallback(
-    (event: React.ChangeEvent<HTMLInputElement>) => {
-      const value = event.target.value;
-      onChange(value, name);
-      setSearchValue(value);
-    },
-    [name, onChange],
-  );
-
-  const handleSelect = useCallback(
-    (id: any) => {
-      id = id ?? value ?? '';
-      const changed = id !== value;
-
-      menu.hide();
-      if (state && changed) {
-        state[name] = id;
-      }
-      if (onSelect && changed) {
-        onSelect(id, name, value);
-      }
-      if (context && changed) {
-        context.change(id, name);
-      }
-      setSearchValue(null);
-    },
-    [value, state, name, menu, context, onSelect],
+  const selectedItem = items.find((item, index) => keySelector(item, index) === selectedKey);
+  const [inputValue, setInputValue] = useState<string | null>(null);
+
+  const selectedValue = selectedItem ? valueSelector(selectedItem) : '';
+  const displayValue = inputValue ?? selectedValue;
+
+  const filteredItems = getComputed(() =>
```

Comment:
`getComputed` will not work as expected for not primitive values, for arrays you will need to specify a custom equality function (I recommend removing `getComputed`)